### PR TITLE
Priority index bug

### PR DESCRIPTION
**Pull request by mqueme**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #2692
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR checks to see if priority and attempts index is set, since it can be missing from campaigns build in older versions

#### Steps to test this PR:
1. If you have an old campaign that sends an email, this PR should remove the priority index error reported by some users

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. if you have an old campaign that sends emails, you might see an error stating that priority or attempts index is not set

### CODE REVIEW

**Comment by mackina** [comment]
On this issue - (i've reported a similar one) did it stop the campaign from executing, or merely report an error?

**Comment by mqueme** [comment]
If its just a notice it should just report the notice, but it should not stop the campaign from executing. From the error you reported, I see it is just a notice, and that is what this PR fixes.

File: app/bundles/FormBundle/Entity/Submission.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ public static function loadMetadata(ORM\ClassMetadata $metadata)
 
         $builder->addIpAddress();
 
-        $builder->addLead(true, 'SET NULL');
+        $builder->addLead(false, 'SET NULL');
```
Comment (alanhartless) [inline app/bundles/FormBundle/Entity/Submission.php]: This changes the table schema. Could you please give some details on the need for this change?

File: app/bundles/FormBundle/Model/SubmissionModel.php [hunk 2]
```diff
@@ -326,18 +326,19 @@ public function saveSubmission($post, $server, Form $form, Request $request = nu
...
             $lead = $this->createLeadFromSubmit($form, $leadFieldMatches, $leadFields);
             $submission->setLead($lead);
-        }
-
-        // Get updated lead if applicable with tracking ID
-        if ($form->isInKioskMode()) {
-            $lead = $this->leadModel->getCurrentLead();
         } else {
-            list($lead, $trackingId, $generated) = $this->leadModel->getCurrentLead(true);
+            // Get updated lead if applicable with tracking ID
```
Comment (alanhartless) [inline app/bundles/FormBundle/Model/SubmissionModel.php]: Could you share reasoning behind this change? It'll cause the tracking ID to not be associated with the submission which breaks A/B tests.

File: app/bundles/CoreBundle/Helper/AbstractFormFieldHelper.php [hunk 3]
```diff
@@ -139,7 +139,7 @@ public static function parseList($list, $removeEmpty = true, $ignoreNumerical =
         $choices = [];
 
         if ($list) {
-            foreach ($list as $val => $label) {
+            foreach ((array) $list as $val => $label) {
```
Comment (alanhartless) [inline app/bundles/CoreBundle/Helper/AbstractFormFieldHelper.php]: Did you hit a notice with this sparking this change and is it reproducible? If so, worries me we have a bug elsewhere and this is just a patch.

Comment (mqueme) [inline app/bundles/CoreBundle/Helper/AbstractFormFieldHelper.php]: I did, but I think it was due to my fields not being correctly.i have tested on new installation at I had no issue. Will remove from this PR

**Comment by escopecz** [comment]
I tested with a legacy campaign and I was able to replicate the warning. This PR fixed it :+1:

**Pull request merged**